### PR TITLE
Merge pull request #1 from webdizz/master

### PR DESCRIPTION
**Pull request by VasylKochetov**
update main

### CODE REVIEW

File: plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/DateFormatUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DateFormatUtilTest {
+	private static final Logger LOGGER = LoggerFactory.getLogger(DateFormatUtilTest.class);
+	private String sampleDate = "10:20:30 December 03, 1999";
```
Comment (webdizz) [inline plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/DateFormatUtilTest.java]: looks like a constant, please make it to look like constant as well

File: plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/DateFormatUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	private String wrongDate = "10 March 2015 12:00:00";
+	private DateTime sampleDateTime = new DateTime (1999, 12, 3, 10, 20, 30);
+	private String invalidFormatMessage = "java.lang.IllegalArgumentException: Invalid format: ";
+	
+	@BeforeClass
```
Comment (webdizz) [inline plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/DateFormatUtilTest.java]: why do we need these empty methods?

File: plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/DateFormatUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	@Before
+	public void setUp() throws Exception {
+	}
+
+	@After
+	public void tearDown() throws Exception {
+	}
+
+	@Test
+	public void testFormatDate() throws Exception {
```
Comment (webdizz) [inline plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/DateFormatUtilTest.java]: test starts with "should" as a narrative of what this test is about to test

File: plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/DateFormatUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		LOGGER.info("Testing format date");
+		assertTrue(DateFormatUtil.formatDate(sampleDateTime).equals(sampleDate));
+		assertFalse(DateFormatUtil.formatDate(sampleDateTime).equals(wrongDate));
+		LOGGER.info("Done.");
```
Comment (webdizz) [inline plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/DateFormatUtilTest.java]: do you really need these logs?

File: plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/EnvVarsResolverTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    public JenkinsRule j = new JenkinsRule();
+	
+	private String environmentKey = "qa";
+	private String buildVersion = "0.8";
+	private String applicationName = "app1";
+	
+	private EnvVarsResolver resolver;
+	
+	@BeforeClass
+	public static void setUpBeforeClass() throws Exception {
```
Comment (webdizz) [inline plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/EnvVarsResolverTest.java]: same as above

File: plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/EnvVarsResolverTest.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+		value.buildEnvVars(build, envVars);
+		
+		build.addAction(new DynamicVariablesStorageAction(envVars));
+		
+		LogTaskListener listener = new LogTaskListener(log.getLogger(Jenkins.class.getName()), Level.INFO);
+		resolver = new EnvVarsResolver(build, listener);
+	}
+
+	@Test
+	public void testGetValue() throws Exception {
```
Comment (webdizz) [inline plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/EnvVarsResolverTest.java]: test methods starts with "should"

File: plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/EnvVarsResolverTest.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+		assertTrue(TEST_BUILD_VERSION.equals(resolver.getValue(Constants.BUILD_VERSION)));
+		assertTrue(TEST_ENV_KEY.equals(resolver.getValue(Constants.ENV_NAME)));
```
Comment (webdizz) [inline plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/EnvVarsResolverTest.java]: use hamcrest assertion

File: plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/EnvVarsResolverTest.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+	
+	private static final String TEST_ENV_KEY = "qa";
+	private static final String TEST_BUILD_VERSION = "0.8";
+	private static final String TEST_APP_NAME = "app1";
+	
+	private EnvVarsResolver resolver;
+
+	@Before
+	public void setUp() throws Exception {
+		FreeStyleProject project = j.createFreeStyleProject("Deployment test job");
```
Comment (webdizz) [inline plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/EnvVarsResolverTest.java:57]: is this set up specific for all tests? if no it's better to encapsulate within test method then extract if any

Comment (VasylKochetov) [inline plugin/src/test/java/com/epam/grandhackathon/deployment/sphere/plugin/utils/EnvVarsResolverTest.java:57]: This set up is used at init stage, and then checked within a test. I think should be kept class constant.

**Pull request closed without merging**